src/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -341,6 +341,128 @@ public BitArray Not()
             return this;
         }
 
+        /*=========================================================================
+        ** Shift all the bit values to right on count bits. The current instance is
+        ** updated and returned.
+        * 
+        ** Exceptions: ArgumentOutOfRangeException if count < 0
+        =========================================================================*/
+        public BitArray RightShift(int count)
+        {
+            if (count <= 0)
+            {
+                if (count < 0)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(count), count, SR.ArgumentOutOfRange_NeedNonNegNum);
+                }
+                Contract.EndContractBlock();
```

Comment:
FYI ```Contract.EndContractBlock();``` isn't necessary anymore, so you can leave it out in future additions. It's used in every other method in this file so I see why you added it though.